apps/server/src/modules/h5p-editor/temporary-file-storage/temporary-file-storage.ts

```diff
@@ -147,19 +83,33 @@ export class TemporaryFileStorage implements ITemporaryFileStorage {
 		expirationTime: Date
 	): Promise<ITemporaryFile> {
 		this.checkFilename(filename);
-		this.checkFilename(user.id);
 		const now = new Date();
 		if (expirationTime < now) {
 			throw new Error('expirationTime must be in the future');
 		}
 
-		const path = join(this.path, user.id, filename);
-		await this.ensureDirExists(dirname(path));
+		const path = join(user.id, filename);
+		let tempFile: TemporaryFile | undefined;
+		try {
+			tempFile = await this.repo.findByPath(user.id, filename);
+			await this.s3Client.delete([path]);
+		} catch (err) {
+			/* does not exist */
+		}
+		await this.s3Client.create(
+			path,
+			new FileDto({ name: path, mimeType: 'application/octet-stream', data: dataStream })
+		);
 
-		await writeFile(path, dataStream);
-		const meta = new TemporaryFile(filename, user.id, expirationTime);
-		await writeFile(`${path}.meta`, JSON.stringify(meta));
+		if (tempFile === undefined) {
+			tempFile = new TemporaryFile(filename, user.id, expirationTime, new Date(), dataStream.bytesRead);
+			await this.repo.createTemporaryFile(tempFile);
```

Comment:
```suggestion
			await this.repo.save(tempFile);
```
The MikroORM docs say "we instantiate new entity via new Author() or em.create(Author, { ... })". (https://mikro-orm.io/docs/defining-entities#default-values) So you don't need to do both in a row. 

You can also get rid of that repo method then, as it is not used anywhere else.

Reply:
Removed `createTemporaryFile` Method